packages/flame/lib/src/game/camera/viewport.dart

```diff
@@ -66,29 +66,26 @@ abstract class Viewport extends Projector {
   /// any viewport.
   ///
   /// You probably don't need to care about this if you are using a viewport.
-  Vector2 get canvasSize;
+  Vector2? canvasSize;
```

Comment:
The `canvasSize` is not known (`null`) before everything was laid out. I could have had `hasLayout` boolean flag instead, but it seems that having the property nullable is more straightforward.

Reply:
do you ever need to check for `viewport.canvasSize == null` though? otherwise it can be just a `late Vector2`, so you don't need the `!` everywhere (just assume it will be there always after layout)